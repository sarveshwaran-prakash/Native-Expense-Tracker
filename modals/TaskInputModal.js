// TaskInputModal.js
import React, { useState } from "react";
import {
  Modal,
  View,
  TextInput,
  Button,
  StyleSheet,
  TouchableOpacity,
  Dimensions,
} from "react-native";
import { FontAwesome } from "@expo/vector-icons";

const { width, height } = Dimensions.get("window");

export default function TaskInputModal({ visible, onClose, onAddTask }) {
  const [title, setTitle] = useState("");
  const [description, setDescription] = useState("");

  const handleAddTask = () => {
    onAddTask({ title, description }); // Pass both title and description as an object
    setTitle("");
    setDescription("");
  };

  return (
    <Modal
      animationType="slide"
      transparent={true}
      visible={visible}
      onRequestClose={onClose}
    >
      <View style={styles.container}>
        <View style={styles.modalView}>
          <TouchableOpacity onPress={onClose} style={styles.closeButton}>
            <FontAwesome name="times" size={24} color="black" />
          </TouchableOpacity>
          <TextInput
            style={styles.input}
            value={title}
            onChangeText={setTitle}
            placeholder="Enter task title"
          />
          <TextInput
            style={[styles.input, styles.descriptionInput]}
            value={description}
            onChangeText={setDescription}
            placeholder="Enter description (optional)"
            multiline={true}
            numberOfLines={4}
          />
          <Button
            title="Add Task"
            onPress={handleAddTask}
            disabled={title.trim() === ""}
          />
        </View>
      </View>
    </Modal>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "rgba(0, 0, 0, 0.5)",
  },
  modalView: {
    width: width * 0.8,
    maxHeight: height * 0.6,
    backgroundColor: "white",
    borderRadius: 20,
    padding: 35,
    alignItems: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  input: {
    width: "100%",
    marginBottom: 10,
    padding: 10,
    borderWidth: 1,
    borderColor: "#ccc",
    borderRadius: 5,
  },
  descriptionInput: {
    height: 100,
  },
  closeButton: {
    position: "absolute",
    top: 10,
    right: 10,
  },
});
