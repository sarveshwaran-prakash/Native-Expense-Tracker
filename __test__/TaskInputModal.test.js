import React from "react";
import renderer from "react-test-renderer";
import TaskInputModal from "../modals/TaskInputModal";

describe("TaskInputModal", () => {
  it("renders correctly when visible", () => {
    const mockOnClose = jest.fn();
    const mockOnAddTask = jest.fn();

    const tree = renderer
      .create(
        <TaskInputModal
          visible={true}
          onClose={mockOnClose}
          onAddTask={mockOnAddTask}
        />
      )
      .toJSON();

    // Ensure the modal elements are rendered
    expect(tree).toMatchSnapshot();

    // Simulate adding a task
    const instance = renderer.create(
      <TaskInputModal
        visible={true}
        onClose={mockOnClose}
        onAddTask={mockOnAddTask}
      />
    ).root;

    const textInput = instance.findByProps({ placeholder: "Enter task title" });
    textInput.props.onChangeText("Test Task");

    const button = instance.findByProps({ title: "Add Task" });
    button.props.onPress();

    // Ensure the onAddTask callback is called with the correct parameters
    expect(mockOnAddTask).toHaveBeenCalledWith({
      title: "Test Task",
      description: "",
    });

    // Ensure the onClose callback is called when the close button is pressed
    const closeButton = instance.findByProps({ name: "times" });
    closeButton.props.onPress();
    expect(mockOnClose).toHaveBeenCalled();
  });
});
