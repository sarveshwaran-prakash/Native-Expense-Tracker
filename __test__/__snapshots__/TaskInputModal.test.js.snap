// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskInputModal renders correctly when visible 1`] = `
<View
  animationType="slide"
  onRequestClose={[MockFunction]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 20,
          "elevation": 5,
          "maxHeight": 800.4,
          "padding": 35,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": 600,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "position": "absolute",
            "right": 10,
            "top": 10,
          }
        }
      >
        <Text />
      </View>
      <TextInput
        onChangeText={[Function]}
        placeholder="Enter task title"
        style={
          {
            "borderColor": "#ccc",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginBottom": 10,
            "padding": 10,
            "width": "100%",
          }
        }
        value=""
      />
      <TextInput
        multiline={true}
        numberOfLines={4}
        onChangeText={[Function]}
        placeholder="Enter description (optional)"
        style={
          [
            {
              "borderColor": "#ccc",
              "borderRadius": 5,
              "borderWidth": 1,
              "marginBottom": 10,
              "padding": 10,
              "width": "100%",
            },
            {
              "height": 100,
            },
          ]
        }
        value=""
      />
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
              {},
            ]
          }
        >
          <Text
            disabled={true}
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                {
                  "color": "#cdcdcd",
                },
              ]
            }
          >
            Add Task
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
